application/common/models/Reset.php

```diff
@@ -272,7 +272,7 @@ private function sendAll()
 
         if (is_iterable($methods)) {
             foreach ($methods as $method) {
-                if ($method['verified']) {
+                if ($method['verified'] ?? false) {
```

Comment:
At this point, would it be worth extracting a `Method::getVerifiedMethods(...)` function?

Reply:
Sure. Just for my own edification, does the idea come from seeing the nested `if`/`foreach`/`if`?

Reply:
To some extent, yes. It just seemed like there was enough logic in this function that it seemed like some of it could be extracted and significantly simplify this method, to something like this:
```php
    /**
     * Send reset messages to primary email and all verified recovery methods.
     */
    private function sendAll()
    {
        $this->sendPrimary();

        $methods = Method::getVerifiedMethods($this->user->employee_id);

        foreach ($methods as $method) {
            $this->sendMethod($method['value']);
        }
    }
```
Maybe it was that it seemed like this method had more than 1 responsibility: it both had to send resets to the appropriate methods, and it had to determine whether a method was verified or not.